Sources/MintKit/Mint.swift

```diff
@@ -84,6 +84,11 @@ public class Mint {
         return gitRepos
     }
 
+    func getDefaultGitBranch() throws -> String {
+        let refsOutput = try Task.capture(bash: "git ls-remote --heads --quiet | cut -f2 | cut -d'/' -f3")
```

Comment:
Are you sure this always works? I tried locally in some repos. Some I got an access error if though they have public remotes:
```
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```
And some I just got a list of the branches in the repo, and not just the default branch. I'm on Git 2.24.3 and zsh 5.8

Reply:
I use this in my zsh function, it's a bit different: `defaultBranch=$(git ls-remote --symref "$remoteUrl" HEAD | grep "^ref: " | sed 's@^ref: refs/heads/\([^[:space:]]*\).*@\1@')`

Reply:
Assuming pipefail is on, my version will result in an error in the rare case where the default branch is not set, while the original code will probably not result in an error. (I'm not on my dev machine, so I can't check but I'm 99% sure about this)

related git docs: https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-headem

Reply:
If it's assume both 1) that the repo was cloned, not created locally, and 2) that the first (or only) remote is the one that matters), this should be as good as it seems to be possible to get within the bounds of what's practical (I know the string extraction is a bit tortured; while either `sed` or the new regex support in Swift 5.7 could do this less painfully, `sed` is not always dependable between platforms, and 5.7 is a bit of a jump from this project's current minimum supported version of 5.0):
```swift
let symref = try Task.capture(bash: "git ls-remote --symref . 'refs/remotes/*/HEAD'")
if symref.starts(with: "ref:") { return symref
    .prefix { !$0.isNewline }    // "ref: refs/remotes/origin/main    refs/remotes/origin/HEAD"
    .dropFirst(17)               // "origin/main    refs/remotes/origin/HEAD"
    .drop { $0 != "/" }          // "/main    refs/remotes/origin/HEAD"
    .dropFirst()                 // "main    refs/remotes/origin/HEAD"
    .prefix { !$0.isWhitespace } // "main"
} else {
    // no remote configured?
    return "HEAD" // fall back to current checked out revision
}
```